File: src/main/java/org/micoli/php/ui/popup/NavigableOpenAllAction.java
package org.micoli.php.ui.popup;

import com.intellij.pom.Navigatable;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public class NavigableOpenAllAction implements NavigableListPopupItem {
    private final List<Navigatable> navigables;

    public NavigableOpenAllAction(List<Navigatable> navigables) {
        this.navigables = navigables;
    }

    @Override
    public void navigate(boolean requestFocus) {
        this.navigables.forEach(navigable -> navigable.navigate(requestFocus));
    }

    @Override
    public boolean canNavigate() {
        return true;
    }

    public boolean canNavigateToSource() {
        return false;
    }

    @NotNull
    public String getText() {
        return String.format(
                "<html><div style=\"padding:5px\"><i style='color: orange;font-weight:bold'>%s</i></div></html>",
                "Open results in Editor");
    }
}

File: src/main/kotlin/org/micoli/php/ui/popup/NavigableOpenAllAction.kt
package org.micoli.php.ui.popup

import com.intellij.pom.Navigatable
import kotlin.Boolean
import kotlin.text.trimIndent
import kotlinx.collections.immutable.ImmutableList

class NavigableOpenAllAction(private val navigables: ImmutableList<Navigatable?>) :
    NavigableListPopupItem {
    override fun navigate(requestFocus: Boolean) {
        this.navigables.filter { it != null }.forEach { it!!.navigate(requestFocus) }
    }

    override fun canNavigate(): Boolean {
        return true
    }

    override fun canNavigateToSource(): Boolean {
        return false
    }

    override fun getText(): String {
        return String.format(
            """
                <html>
                    <div style="padding:5px">
                        <i style="color: orange;font-weight:bold">%s</i>
                    </div>
                </html>,
                
                """
                .trimIndent(),
            "Open results in Editor",
        )
    }
}
